### PR TITLE
Skeleton for catalogue items and attachments

### PR DESCRIPTION
**Pull request by mattdean-digicatapult**
Currently contains skeleton for:

* `attachment` API
* `catalogue-item` API
* `build` API
* `order` API
* `part` API

Skeleton consists of:

* path declarations and descriptions
* request schema
* response schema

### CODE REVIEW

File: app/api-v1/api-doc.js [hunk 1]
```diff
@@ -26,10 +26,463 @@ const apiDoc = {
...
+          owner: {
+            description:
+              'Name of the submitter of the purchase-order. This information is not stored directly on-chain',
+            type: 'string',
+            maxLength: 255,
+          },
+          status: {
+            type: 'string',
+            description: 'Status of the purchase-order',
+            enum: ['Created', 'Submitted', 'Rejected', 'Accepted'],
```
Comment (jonmattgray) [inline app/api-v1/api-doc.js:292]: + amended

File: app/api-v1/api-doc.js [hunk 2]
```diff
@@ -26,10 +26,463 @@ const apiDoc = {
...
+            description:
+              'Name of the OEM who owns the design of the catalogue-item. This information is not stored directly on-chain',
+            type: 'string',
+            maxLength: 255,
+          },
+        },
+      },
+      CertificationRequirement: {
+        type: 'object',
+        properties: {
```
Comment (jonmattgray) [inline app/api-v1/api-doc.js:91]: for `inteli-demo`, certificates also have a `metadataKey` attribute so we know which metadata key to assign an uploaded cert file to on-chain
https://github.com/digicatapult/inteli-demo/blob/9b2e12866d8abbc7c07df639ba9c800fd4296f58/src/features/partsSlice.js#L4

Comment (jonmattgray) [reply app/api-v1/api-doc.js:91]: we could generate the required keys though (and guarantee they're unique)

File: app/api-v1/api-doc.js (hunk 1 above)
Comment (mattdean-digicatapult) [reply app/api-v1/api-doc.js]: done

File: app/api-v1/api-doc.js (hunk 2 above)
Comment (mattdean-digicatapult) [reply app/api-v1/api-doc.js:91]: I think we can auto-generate this

**Review by jonmattgray — APPROVED** [review]

**Pull request merged**